src/Raven.Studio/typescript/viewmodels/database/query/query.ts

```diff
@@ -1168,6 +1179,45 @@ class query extends viewModelBase {
         });
     }
     
+    private onSpatialLoaded(queryResults: pagedResultExtended<document>) {
+        const spatialProperties = queryResults.additionalResultInfo.SpatialProperties;
+        if (spatialProperties && queryResults.items.length) {
+            this.isSpatialQuery(true);
+            this.spatialDataFound(true);
+
+            // Each spatial map model will contain the layer of markers per spatial properties pair
+            let spatialMapModels: spatialMapLayerModel[] = [];
+
+            for (let i = 0; i < spatialProperties.length && this.spatialDataFound(); i++) {
+                const latitudeProperty = spatialProperties[i].LatitudeProperty;
+                const longitudeProperty = spatialProperties[i].LongitudeProperty;
+
+                let pointsArray: geoPoint[] = [];
+                for (let i = 0; i < queryResults.items.length; i++) {
+                    const item = queryResults.items[i];
+
+                    const latitudeValue = _.get(item, latitudeProperty) as number;
+                    const longitudeValue = _.get(item, longitudeProperty) as number;
+                    this.spatialDataFound(!!latitudeValue && !!longitudeValue);
```

Comment:
what if all values are (0,0)? Do we need this spatial data found?

Reply:
`spatialDataFound `removed.
Only markers that are found are pushed to the array.